src/spring-cloud/azext_spring_cloud/_validators.py

```diff
@@ -1,438 +1,542 @@
-    if not name:
-        return
-    matchObj = match(r'^[-\w\._\(\)]+$', name)
-    if matchObj is None:
-        raise CLIError('--{0} must conform to the following pattern: \'^[-\\w\\._\\(\\)]+$\'.'.format(message_name))
-
-
-def _validate_route_table(namespace, vnet_obj):
-    app_route_table_id = ""
-    runtime_route_table_id = ""
-    for subnet in vnet_obj.subnets:
-        if subnet.id.lower() == namespace.app_subnet.lower() and subnet.route_table:
-            app_route_table_id = subnet.route_table.id
-        if subnet.id.lower() == namespace.service_runtime_subnet.lower() and subnet.route_table:
-            runtime_route_table_id = subnet.route_table.id
-
-    if app_route_table_id and runtime_route_table_id:
-        if app_route_table_id == runtime_route_table_id:
-            raise CLIError('--service-runtime-subnet and --app-subnet should associate with different route tables.')
-    if (app_route_table_id and not runtime_route_table_id) \
-            or (not app_route_table_id and runtime_route_table_id):
-        raise CLIError(
-            '--service-runtime-subnet and --app-subnet should both associate with different route tables or neither.')
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=too-few-public-methods, unused-argument, redefined-builtin
+
+from re import match
+from re import search
+from ipaddress import ip_network
+import zipfile
+from azure.cli.core import telemetry
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core.commands.validators import validate_tag
+from azure.cli.core.util import CLIError
+from msrestazure.tools import is_valid_resource_id
+from msrestazure.tools import parse_resource_id
+from msrestazure.tools import resource_id
+from knack.log import get_logger
+from ._utils import ApiType
+from ._utils import _get_rg_location
+
+logger = get_logger(__name__)
+
+
+def validate_env(namespace):
+    """ Extracts multiple space-separated envs in key[=value] format """
+    if isinstance(namespace.env, list):
+        env_dict = {}
+        for item in namespace.env:
+            env_dict.update(validate_tag(item))
+        namespace.env = env_dict
+
+
+def validate_location(namespace):
+    if namespace.location:
+        location_slice = namespace.location.split(" ")
+        namespace.location = "".join([piece.lower()
+                                      for piece in location_slice])
+
+
+def validate_sku(namespace):
+    if namespace.sku is not None:
+        namespace.sku = namespace.sku.upper()
+        if namespace.sku not in ['BASIC', 'STANDARD']:
+            raise CLIError("The pricing tier only accepts value [Basic, Standard]")
```

Comment:
Could you the specific error type instead of `CLIError`? Such as: `InvalidArgumentValueError`

Reply:
fixed